### ISSUE TITLE
Segmentation fault on macOS 11/arm64 (M1)

### ISSUE THREAD

**Issue by shawnanastasio** [Legrandin/pycryptodome#506]
When running the `pycryptodomex==3.10.1` test suite on an M1 mac, CPython segfaults:
```
$ python -m Cryptodome.SelfTest
Skipping AESNI tests
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Cipher/test_DES3.py:60: UserWarning: Warning: skipping extended tests for TDES ECB (TECBMMT2.rsp)
  test_vectors = load_test_vectors(
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Cipher/test_DES3.py:60: UserWarning: Warning: skipping extended tests for TDES ECB (TECBMMT3.rsp)
  test_vectors = load_test_vectors(
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Cipher/test_GCM.py:782: UserWarning: Warning: skipping extended tests for GCM decrypt
  test_vectors_nist = load_test_vectors(
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Cipher/test_GCM.py:788: UserWarning: Warning: skipping extended tests for GCM encrypt
  test_vectors_nist += load_test_vectors(
Skipping test of PCLMULDQD in AES GCM
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA1.py:58: UserWarning: Warning: skipping extended tests for KAT SHA-1
  test_vectors = load_test_vectors(("Hash", "SHA1"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA3_224.py:59: UserWarning: Warning: skipping extended tests for KAT SHA-3 224
  test_vectors = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA3_256.py:59: UserWarning: Warning: skipping extended tests for KAT SHA-3 256
  test_vectors = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA3_384.py:59: UserWarning: Warning: skipping extended tests for KAT SHA-3 384
  test_vectors = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA3_512.py:59: UserWarning: Warning: skipping extended tests for KAT SHA-3 512
  test_vectors = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:144: UserWarning: Warning: skipping extended tests for Short Messages KAT 224
  test_vectors_224 =  load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:149: UserWarning: Warning: skipping extended tests for Long Messages KAT 224
  test_vectors_224 += load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:168: UserWarning: Warning: skipping extended tests for Short Messages KAT 256
  test_vectors_256 =  load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:173: UserWarning: Warning: skipping extended tests for Long Messages KAT 256
  test_vectors_256 += load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:193: UserWarning: Warning: skipping extended tests for Short Messages KAT 384
  test_vectors_384 =  load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:198: UserWarning: Warning: skipping extended tests for Long Messages KAT 384
  test_vectors_384 += load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:217: UserWarning: Warning: skipping extended tests for Short Messages KAT 512
  test_vectors_512 =  load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_keccak.py:222: UserWarning: Warning: skipping extended tests for Long Messages KAT 512
  test_vectors_512 += load_test_vectors(("Hash", "keccak"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHAKE.py:94: UserWarning: Warning: skipping extended tests for Short Messages KAT SHAKE128
  test_vectors_128 = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHAKE.py:113: UserWarning: Warning: skipping extended tests for Short Messages KAT SHAKE256
  test_vectors_256 = load_test_vectors(("Hash", "SHA3"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA512.py:55: UserWarning: Warning: skipping extended tests for KAT SHA-512
  test_vectors = load_test_vectors(("Hash", "SHA2"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA512.py:79: UserWarning: Warning: skipping extended tests for KAT SHA-512/224
  test_vectors = load_test_vectors(("Hash", "SHA2"),
/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/SelfTest/Hash/test_SHA512.py:104: UserWarning: Warning: skipping extended tests for KAT SHA-512/256
  test_vectors = load_test_vectors(("Hash", "SHA2"),
Segmentation fault: 11
```

Running with -v shows that the `Random` submodule may be at fault:
```
...

# code object from '/Volumes/shawnanastasio_home/opt/pycryptodome/debug/lib/python3.9/site-packages/Cryptodome/Random/__pycache__/random.cpython-39.pyc'
import 'Cryptodome.Random.random' # <_frozen_importlib_external.SourceFileLoader object at 0x101a74880>
Segmentation fault: 11
```

**Comment by Legrandin** [comment]
How did you install Python 3.9?
I can't reproduce but I only tried with the stock Python 3.8.

**Comment by aelgohar** [comment]
Same issue here. Did you manage to fix it?

**Comment by shawnanastasio** [comment]
> How did you install Python 3.9?
> I can't reproduce but I only tried with the stock Python 3.8.

I'm using python 3.9 from Homebrew, sorry, I should have specified that in the original report.

> Same issue here. Did you manage to fix it?

I haven't looked into the issue since creating this ticket, but it's still present for me.

**Comment by aelgohar** [comment]
Python version seems to be the issue. Works fine on python 3.8.

**Comment by Legrandin** [comment]
I installed python 3.9.1 with brew 3.0.1 and I can compile and successfuly test.

Can you clarify:
* exact version of Python 3.9 (3.9.0 or 3.9.1?)
* exact version of brew
* whether you see this problem with Rosetta or on native Arm64

**Comment by aelgohar** [comment]
Python 3.9.1
brew 3.0.1
native arm64, running Big Sur 11.2.1

**Comment by seisvelas** [comment]
I get the exact same error - Python 3.9 on M1 mac

**Comment by jameskirsop** [comment]
Tangentially related, but I'm surprised you're getting this far at all. I'm getting wheel build issues on an M1 Mac running Python3.9, and version 3.9.9 of this module:
```
...
1.) The wheel tags for pycryptodome 3.9.9 are cp39-cp39-macosx_10_9_universal2 which do not match the supported tags of DistributionTarget(interpreter=PythonInterpreter('/Library/ManagedFrameworks/Python/Python3.framework/Versions/3.9/bin/python3.9', PythonIdentity('/usr/local/bin/managed_python3', 'cp39', 'cp39', 'macosx_11_0_arm64', (3, 9, 1)))):
...
```

**Comment by Legrandin** [comment]
As an update, I plan to fix this soon.
As a workaround, it seems that installing `cffi` does not trigger the problem.

**Comment by Legrandin** [comment]
Fixed in version 3.10.3.
